streams/src/main/java/org/apache/kafka/streams/errors/ErrorHandlerContextImpl.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.streams.errors;
```

Comment:
The implementation should go into an internal package since it is an implementation detail that should not be exposed to the public. Otherwise, we would need a KIP for it.
You can create a package  `org.apache.kafka.streams.errors.internals`.

Reply:
Done. Should `DefaultErrorHandlerContext` class be `public` in this PR? We don't need it to be public in this PR, but we would need to access it from `org.apache.kafka.streams.processor` in the next PRs

Reply:
`public` is fine.